### DISCUSSION TITLE
cut/copy/paste issues

### DISCUSSION THREAD

**Discussion by mgutz** [hyprwm/Hyprland#4834] [General]
I'm having sporadic issues with cut/copy/paste across applications. Most of the time it works, but, as an example, I'll have to copy the URL many times from Firefox address bar to paste it into foot or obsidian. Does not seem to matter if copy is done with the mouse or keyboard. It often takes many retries. 

This happens intermittently between VS Code, Firefox, foot and obsidian all running in wayland mode (verified with xwininfo). I'm not running any clipboard utility.

Anybody else running into this?

**Comment by varris1** [comment]
Are you trying to paste things from wayland into xwayland windows? If you do, there is a solution inside https://github.com/hyprwm/Hyprland/issues/2319

**Comment by rsacherer** [comment]
Yeah, *suddenly* ran into this too, never had this issue, then suddenly it popped up, it might have to do me switching from firefox to chromium and it being an xwayland app.

It worked when copying from xwayland chrome, pasting into foot, copying from foot and pasting into obsidian for example. 

Also I had two chromium instances to separate work flows (one in a scratch window for communication, the other for work) and one of those was wayland, the other xwayland.

I now reconfigured obsidian to run native wayland and chromium as well, so now nearly all apps I run are pure wayland and it seems much better.

**Comment by XenGi** [comment]
I'm having these issues for a while now too. Happens randomly between applications. Sometimes only in one direction. I had a flatpak app that I gave access to wayland and then it worked but that didn't work in other flatpaks.
As of now it seems I can't copy between xwayland and wayland. xwayland to xwayland seems to work and also wayland to wayland.

**Comment by TheKoTech** [comment]
It sucks that such basic functionality barely works. From my experience, it's a lot worse with flatpaks.

I'm using `clipse`, copying in xwayland windows just doesn't get saved. Xwayland to xwayland is fine, Wayland to xwayland is also fine, but then I can't paste a url from chrome to telegram or some other wayland app.

Really annoying.

**Comment by Polyhistor** [comment]
Same issue here

**Comment by 1096015776** [comment]
hey, i have same issue in here. but I found other method to solve it.maybe you just need a xwayland-satellite. as for me, clip it can work well,but tomorrow , it not work. i guess it's xwayland mulitpe instance let it bad.so i try  xwayland-satellite, now clip work well.
